invenio_records_resources/resources/record.py

```diff
@@ -49,55 +67,49 @@ def search(self):
         }
         pagination = request_args
 
+        # TODO: Args parsing has to allow for easily adding new parameters,
+        # when someone extends the search engine.
```

Comment:
- SearchURLArgsSchema() does this for URL search args, I think... i.e perhaps passing `request_args` and letting search_engine deal with it?